src/Assets/Javascript/scroll-tracking.js

```diff
@@ -0,0 +1,49 @@
+import throttle from "lodash.throttle";
+
+function ScrollTracking($module) {
+  this.$module = $module;
+}
+
+const triggerAnalyticsEvent = (category, action) => {
+  ga("send", "event", {
+    eventCategory: category,
+    eventAction: action,
+    transport: "beacon",
+    nonInteraction: true
+  });
+};
+
+function onScroll() {
+  const h = document.documentElement;
+  const b = document.body;
+  const st = "scrollTop";
+  const sh = "scrollHeight";
+
+  let scrolledPastOneThird = false;
+  let scrolledPastTwoThirds = false;
+  let scrolledToBottom = false;
```

Comment:
These variables are initialised with `false` every time the `onScroll` function runs, but we need to persist them between calls to the function; otherwise they make no effective difference to the conditionals.

Because we're using webpack modules, and this is a module which is meant to run only once per page, and our app is a multi-page server rendered app (so the JS state refreshes on every page navigation), we can chuck them anywhere in the scope of this file, outside this function, without any ill effects:

```js
let scrolledPastOneThird = false;
let scrolledPastTwoThirds = false;
let scrolledToBottom = false;

function onScroll() {
  // ...
}
```

If we expect there to be multiple scroll modules, we would need to use some kind of closure + factory:

```js
function onScrollFactory() {
  let scrolledPastOneThird = false;
  let scrolledPastTwoThirds = false;
  let scrolledToBottom = false;
  
  return function onScroll() {
    // ...
  }
}

// Then later on, we call the factory:
const onScroll = onScrollFactory();
document.addEventListener("scroll", throttle(onScroll, 250), false);
```

The latter is somewhat more robust but not necessary in this particular usecase I feel.